Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/GUIGamePanel.java

```diff
@@ -0,0 +1,134 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+import javax.swing.JPanel;
+
+@SuppressWarnings("serial")
+public class GUIGamePanel extends JPanel {
+	private String pressedKey;
+	private int hangmanStage;
+
+	public GUIGamePanel(){
+		super();
+		KeyListener listener = new KeyListener() {
+			@Override
+			public void keyTyped(KeyEvent e) {
+			}
+
+			@Override
+			public void keyPressed(KeyEvent e) {
+				//System.out.println("keyPressed="+KeyEvent.getKeyText(e.getKeyCode()));
+				String key=KeyEvent.getKeyText(e.getKeyCode());
+				switch (key){
+				case "Shift":
+				case "Windows":
+				case "Ctrl":
+				case "Alt":
+				case "Caps Lock":
+				case "Space":
+				case "Comma":
+				case "0":
+				case "1":
+				case "2":
+				case "3":
+				case "4":
+				case "5":
+				case "6":
+				case "7":
+				case "9":
+				case "8":
+				case "Minus":
+				case "Equals":
+				case "Backspace":
+				case "Open Bracket":
+				case "Close Bracket":
+				case "Semicolon":
+				case "Quote":
+				case "Slash":
+				case "Back Slash":
+				case "Alt Graph":
+				case "Context Menu":
```

Comment:
Two things.
1) Can you guarantee that this is a universally complete list of non-acceptable keys? Across all languages and keyboard layouts, and going into the future? I would have thought that it would be better to check if the symbol is alphanumeric - you could use if (key.length() == 1 && Character.isAlphabetic(key.charAt(0)) {...} instead.

2) I'm not 100% certain this is the responsibility of the GUI panel. Probably want to put these checks within the Game logic, as no matter where the input comes from, the Game itself defines what is an acceptable and what is not an acceptable input.